aws/table_aws_ses_domain_identity.go

```diff
@@ -0,0 +1,186 @@
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "arn",
+				Description: "The ARN of the AWS SES identity.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getDomainIdentityARN,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "verification_status",
+				Description: "The verification status of the identity.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getDomainIdentityVerificationAttributes,
+				Transform:   transform.FromField("VerificationStatus"),
+			},
+			{
+				Name:        "verification_token",
+				Description: "The verification token for a domain identity.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getDomainIdentityVerificationAttributes,
+				Transform:   transform.FromField("VerificationToken"),
+			},
+			{
+				Name:        "notification_attributes",
+				Description: "Represents the notification attributes of an identity.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getDomainIdentityNotificationAttributes,
+				Transform:   transform.FromValue(),
+			},
+
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getDomainIdentityARN,
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSESDomainIdentities(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("listSESDomainIdentities")
+	region := d.KeyColumnQualString(matrixKeyRegion)
+
+	// Create Session
+	svc, err := SESService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	input := &ses.ListIdentitiesInput{
+		MaxItems:     aws.Int64(1000),
+		IdentityType: aws.String(ses.IdentityTypeDomain),
+	}
+
+	// Limiting the results
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < *input.MaxItems {
+			if *limit < 1 {
+				input.MaxItems = types.Int64(1)
+			} else {
+				input.MaxItems = limit
+			}
+		}
+	}
+
+	// List call
+	err = svc.ListIdentitiesPages(
+		input,
+		func(page *ses.ListIdentitiesOutput, lastPage bool) bool {
+			for _, identity := range page.Identities {
+				d.StreamListItem(ctx, *identity)
+
+				// Context can be cancelled due to manual cancellation or the limit has been hit
+				if d.QueryStatus.RowsRemaining(ctx) == 0 {
+					return false
+				}
+			}
+			return !lastPage
+		},
+	)
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getDomainIdentityVerificationAttributes(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
For this hydrate function and others, the code looks similar/identical to those defined for the `aws_ses_email_identity` table.

I think in this case, we can re-use the hydrate functions from `table_aws_ses_email_identity.go`. and can also rename them as well to be more generic, e.g., `getEmailIdentityVerificationAttributes` -> `getSESIdentityVerificationAttributes`, `getEmailIdentityNotificationAttributes` -> `getSESIdentityNotificationAttributes`.

In this table file, then the only change you'd need to make is updating references to those files, e.g.,
```
{
	Name:        "verification_token",
	Description: "The verification token for a domain identity.",
	Type:        proto.ColumnType_STRING,
	Hydrate:     getSESIdentityVerificationAttributes,
	Transform:   transform.FromField("VerificationToken"),
},
```

Reply:
Good point, I was actually thinking about this when copying the code from email_identity, but not sure if tables should "reuse" hydrate functions between them. Renamed the hydrate functions and updated both tables.

Reply:
Good point, I was actually thinking about this when copying the code from email_identity, but not sure if tables should "reuse" hydrate functions between them. Renamed the hydrate functions and updated both tables.